src/inboxsdk-js/inboxsdk-NONREMOTE.ts

```diff
@@ -1,6 +1,8 @@
-import shimLoadScript from './loading/load-platform-implementation-DEV';
+import shimLoaderLoadScript from './loading/load-platform-implementation-NONREMOTE';
 import loader from './loading/platform-implementation-loader';
 
-loader._loadScript = shimLoadScript(500);
+loader._loadScript = shimLoaderLoadScript(500);
 
-export default require('./inboxsdk');
+import * as SDK from './inboxsdk';
```

Comment:
I have a feeling the spec is being violated here. `import`, the keyword as opposed to the function `import()`, is meant to be statically resolved before the rest of the module loads (including line 4). Because of this, with one of the example extensions, it seems I'm seeing:

> loadScript is not available in the NPM version of the InboxSDK. This function is not usable in Chrome MV3 extensions. If you need this and you're not using an MV3 extension, use the NPM module <TODO> instead.

and

```
Uncaught ReferenceError: InboxSDK is not defined
    at content.js:5:1
```

To repro, try

```
yarn gulp clean
yarn gulp
```

with the thread-row example extension.

Reply:
Removing the CJS (which is squirrelly) would be nice though if we can come up with an interface that works with ESM exclusively.

Reply:
I think the preload method is called synchronously when the sdk module is loaded before our monkey-patch supplies which calls load script. I'm thinking of either moving the preload behind a setTimeout(..., 0) or making it be called right after we do the monkey-patch. I'll get this fixed soon. Sorry for pinging you on this before I put this through proper testing.

Reply:
> I think the preload method is called synchronously when the sdk module is loaded which calls load script before our monkey-patch runs. I'm thinking of either moving the preload behind a setTimeout(..., 0) or making it be called right after we do the monkey-patch.


An alternative to `setTimeout` would be to lean on async a bit more. https://github.com/InboxSDK/InboxSDK/pull/971 shows how we might.

> Sorry for pinging you on this before I put this through proper testing.

No big.